temporalcli/commands.workflow_reset_test.go

```diff
@@ -207,3 +214,283 @@ func (s *SharedServerSuite) TestWorkflow_Reset_ToEventID() {
 	s.Equal(1, oneExecutions, "Should not have re-executed the first activity")
 	s.Equal(2, twoExecutions, "Should have re-executed the second activity")
 }
+
+func (s *SharedServerSuite) TestBatchResetByBuildId() {
+	sut := newSystemUnderTest(s)
+
+	sut.startWorkerFor(originalWorkflow, workflowOptions{name: "wf", version: "v1"})
+	sut.executeWorkflow("wf")
+	sut.waitWorkflowBlockedAfterFirstActivity()
+	sut.stopWorkerFor("v1")
+
+	sut.startWorkerFor(extendedWorkflowWithBuggyActivity, workflowOptions{name: "wf", version: "v2"})
+	sut.allowWorkflowToContinue()
+	sut.waitBadActivityExecuted()
+	sut.stopWorkerFor("v2")
+
+	sut.startWorkerFor(extendedWorkflowWithNonDeterministicFix, workflowOptions{name: "wf", version: "v3"})
+	sut.waitBlockOnNonDeterministicError()
+
+	query := fmt.Sprintf("WorkflowId = \"%s\"", sut.run.GetID())
+	s.CommandHarness.Stdin.WriteString("y\n")
+	res := s.Execute(
+		"workflow", "reset",
+		"--address", s.Address(),
+		"--reason", "test-reset-event-id",
+		"--type", "BuildId",
+		"--build-id", sut.buildPrefix+"v2",
+		"--query", query,
+	)
+	require.NoError(s.T(), res.Err)
+
+	sut.assertWorkflowComplete()
+	sut.assertOnlySecondActivityRetried()
+	sut.stopWorkerFor("v3")
+}
+
+const (
+	badActivity = iota
+	firstActivity
+	secondActivity
+	thirdActivity
+)
+
+type assertions interface {
+	NoError(error, ...interface{})
+	Error(error, ...interface{})
+	True(bool, ...interface{})
+	Eventually(func() bool, time.Duration, time.Duration, ...interface{})
+}
+
+type batchResetTestData struct {
+	t           *testing.T
+	buildPrefix string
+	assert      assertions
+	client      client.Client
+	tq          string
+	counters    []atomic.Int32
+	run         client.WorkflowRun
+	ctx         context.Context
+	ctxCancelF  context.CancelFunc
+	workers     map[string]worker.Worker
+	namespace   string
+}
+
+func newSystemUnderTest(suite *SharedServerSuite) batchResetTestData {
+	counters := make([]atomic.Int32, 4)
+	sut := batchResetTestData{
+		t:         suite.T(),
+		counters:  counters,
+		assert:    suite,
+		client:    suite.Client,
+		namespace: suite.Namespace(),
+	}
+	sut.tq = suite.T().Name()
+	sut.buildPrefix = uuid.NewString()[:6] + "-"
```

Comment:
It's just test code so it doesn't matter much, but a bit inconsistent on which fields you choose to set in the struct literal and which after.